### PR TITLE
Only render v3 published embeds #10201

### LINKED ISSUE #10201: maps should not published by default
- create a map
- open it in a incognito window 

it shows the embed and it should not show until it's published

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #10201.

- [x] Handle existing visualizations: #10215.
- [x] Only render v3 published.
- [x] Handle migrations.

#10215 should be run first.

### CODE REVIEW

**Comment by javitonino** [comment]
As well as doing this change, I think we need to autopublish visualizations when migrating to builder, which happens in two ways:
 - Automatically: `Builder::VisualizationsController.auto_migrate_visualization_if_possible` (uses Carto model)
 - By the user: `Api::Json::VisualizationsController.update(version: 3)` (uses old model, maybe look at `Visualization::Member.version=`).

I think it's worth to automatically mapcap old visualizations when they are migrated, as otherwise, as soon as the visualization is migrated, the old embed will start redirecting to the new one which will 404 with this.

Of course, subject to discussion.

**Comment by juanignaciosl** [comment]
@javitonino agreed. I'll handle it in this same PR for consistency. I'll do both. I'll do it after the final confirmation (see [this](https://github.com/CartoDB/cartodb/issues/10201#issuecomment-254434842)).

In addition, do you agree upon not publishing private visualizations? (see [this](https://github.com/CartoDB/cartodb/pull/10215#issuecomment-254429939)).

**Comment by juanignaciosl** [comment]
@javitonino CR this, please. The migration change is not _that_ trivial. I'm getting some unexpected errors at CI as well, I'm not 100%  sure about this but I can't imagine how this code might cause those errors.

**Comment by juanignaciosl** [comment]
@javitonino onpremise tests failed because of "PhantomJS timed out", but both Backend and Frontend ones are ok, so it should be safe.

File: lib/assets/javascripts/cartodb3/editor.js [hunk 1]
```diff
@@ -271,7 +271,10 @@ var mapCreation = function () {
 };
 
 if (visDefinitionModel.get('version') === 2 && modals) {
-  visDefinitionModel.bind('change:version', mapCreation, this);
+  visDefinitionModel.bind('change:version', function () {
```
Comment (javitonino) [inline lib/assets/javascripts/cartodb3/editor.js]: Question for a frontender (@mariacheca): Does this run after getting the response from the PUT to visualizations? Pardon my ignorance, not sure how these events work in Backbone.

**Comment by juanignaciosl** [comment]
Public profile list is based on [a `Visualization::Collection` search](https://github.com/CartoDB/cartodb/blob/1e03fb5014137122fd6fc588c75566b6bda47ca8/app/controllers/admin/pages_controller.rb#L394). As filtering unpublished is not trivial I prefer releasing this ASAP and fixing the listing afterwards, do you agree?

If you do, I'll merge once @MariaCheca confirms your question.

File: lib/assets/javascripts/cartodb3/editor.js (hunk 1 above)
Comment (MariaCheca) [inline lib/assets/javascripts/cartodb3/editor.js]: Yes, this isn't triggered until the update of the visualization is complete and successful. So don't worry, it won't give us any problem :P

**Pull request merged**